src/inmanta/protocol/rest/server.py

```diff
@@ -121,42 +125,60 @@ def respond(self, body: Optional[JsonType], headers: Dict[str, str], status: int
+
             finally:
                 try:
                     await self.finish()
+                except iostream.StreamClosedError:
+                    # The connection has been closed already.
+                    pass
                 except Exception:
                     LOGGER.exception("An exception occurred responding to %s", self.request.remote_ip)
                 self._transport.end_request()
 
     async def head(self, *args: str, **kwargs: str) -> None:
         if args:
             raise Exception("Only named groups are support in url patterns")
-        await self._call(http_method="HEAD", call_config=self._get_config("HEAD"), kwargs=kwargs)
+        await self._transport.add_background_task(
+            self._call(http_method="HEAD", call_config=self._get_config("HEAD"), kwargs=kwargs), cancel_on_stop=False
+        )
 
     async def get(self, *args: str, **kwargs: str) -> None:
         if args:
             raise Exception("Only named groups are support in url patterns")
-        await self._call(http_method="GET", call_config=self._get_config("GET"), kwargs=kwargs)
+        await self._transport.add_background_task(
+            self._call(http_method="GET", call_config=self._get_config("GET"), kwargs=kwargs), cancel_on_stop=False
+        )
 
     async def post(self, *args: str, **kwargs: str) -> None:
         if args:
             raise Exception("Only named groups are support in url patterns")
-        await self._call(http_method="POST", call_config=self._get_config("POST"), kwargs=kwargs)
+        await self._transport.add_background_task(
+            self._call(http_method="POST", call_config=self._get_config("POST"), kwargs=kwargs), cancel_on_stop=False
+        )
 
     async def delete(self, *args: str, **kwargs: str) -> None:
         if args:
             raise Exception("Only named groups are support in url patterns")
-        await self._call(http_method="DELETE", call_config=self._get_config("DELETE"), kwargs=kwargs)
+        await self._transport.add_background_task(
+            self._call(http_method="DELETE", call_config=self._get_config("DELETE"), kwargs=kwargs), cancel_on_stop=False
+        )
 
     async def patch(self, *args: str, **kwargs: str) -> None:
         if args:
             raise Exception("Only named groups are support in url patterns")
-        await self._call(http_method="PATCH", call_config=self._get_config("PATCH"), kwargs=kwargs)
+        await self._transport.add_background_task(
+            self._call(http_method="PATCH", call_config=self._get_config("PATCH"), kwargs=kwargs), cancel_on_stop=False
+        )
 
     async def put(self, *args: str, **kwargs: str) -> None:
         if args:
             raise Exception("Only named groups are support in url patterns")
-        await self._call(http_method="PUT", call_config=self._get_config("PUT"), kwargs=kwargs)
+        await self._transport.add_background_task(
```

Comment:
is cancel ever called before idle_event?

is somewhat confusing as the task added to add_background_task is not a background task at all, but effectively awaited

Reply:
Here it is added so an inflight request can be awaited or cancelled before killing the server.

Reply:
idle_event is already there to await termination of all requests and as far as I can tell, these specific background tasks are only cancelled after idle_event fires

Reply:
These tasks are awaited. And indeed in this case it should not do anything anymore. I added this as a reminder we should handle this better because either they terminate or they are shot down my the final kill.

Reply:
See ticket #1147